[Soh Je Hou] iP

jehousoh opened the pull request:
# **Phil by Duke.™**

> “Sometimes it pays to stay in bed on Monday, rather than spending the rest of the week debugging Monday’s code.” – Dan Salomon

Phil is a buddy that helps remember tasks that you have to perform. It is

- text-based
- easy to learn
- fast to use

Here's how to use it!

1. Download the file [here](https://github.com/jehousoh/ip/releases/tag/A-Jar)
2. Double-click the .jar file
3. Feed Phil with your tasks
4. Relax~

At the cost of only ~~$4.99~~ $0 (For a unlimited time only)!!! 🔥 

Features:

- [x] Managing tasks
- [x] Loading tasks 
- [ ] Reminders (Coming soon)


If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` java
public static void main(String[] args) {
        new Duke("./data/duke.txt").run();
}
```

Commit:
Add Gradle support

jehousoh committed:
added level-1

jehousoh committed:
added level-2

jehousoh committed:
added level-3

jehousoh committed:
Added Todo, Deadline and Event classes and updated toString of Task

jehousoh committed:
added level-4

jehousoh committed:
Added class descriptions

jehousoh committed:
Fixed spacing in Event toString

jehousoh committed:
Added A-TextUiTesting

jehousoh committed:
Added exception handling

jehousoh committed:
Updated UI test

jehousoh committed:
Added descriptions for exception classes

jehousoh committed:
Added level-6

jehousoh committed:
Removed pause from runtest

jehousoh committed:
Added more testcases

jehousoh committed:
Added enum

jehousoh committed:
Remove indentation before "case:"

jehousoh committed:
Add ability to save to .txt

jehousoh committed:
Add understanding of date and time in deadline tasks

jehousoh committed:
Merge branch 'branch-Level-7'

jehousoh committed:
Merge branch 'branch-Level-8'

jehousoh committed:
Bug fix for deadline with no date input

jehousoh committed:
Add more test cases

jehousoh committed:
Change way file is saved

jehousoh committed:
Add Storage system

jehousoh committed:
Add clear list function

jehousoh committed:
Update test code

jehousoh committed:
Add Parser class

jehousoh committed:
Add packages

jehousoh committed:
Clean up packages

jehousoh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jehousoh committed:
Add gradle support

jehousoh committed:
Change createTask function name

jehousoh committed:
Add JUnit tests for duke.task package

jehousoh committed:
Fix file creation

jehousoh committed:
Add JavaDocs

jehousoh committed:
Edit code based on coding standard

jehousoh committed:
Add find function

jehousoh committed:
Merge branch 'branch-A-JavaDoc'

jehousoh committed:
Merge branch 'branch-A-CodingStandard'

jehousoh committed:
Fix merge conflict

jehousoh committed:
Merge branch 'branch-Level-9'

jehousoh committed:
Add JavaDocs

jehousoh committed:
Add JUnit tests

jehousoh committed:
Add javadocs to constructors and tostring

jehousoh committed:
Add checkstyle and styled the code

jehousoh committed:
Add GUI support

alvinjiang1 commented on src/main/java/duke/task/TaskCreator.java:
```diff
@@ -0,0 +1,147 @@
+        int indexOfSplit;
+        String description;
+
+        switch (type) {
+        case TODO:
+            task = new Todo(info);
+            break;
```
I like how you kept to the coding standard for switch cases!

alvinjiang1 commented on src/test/java/duke/task/TaskCreatorTest.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Test
+    public void createDeadlineTest() {
+        assertEquals(TaskCreator.createTask("deadline return book /by 2022-5-12 1800").toString(),
+                "[D][ ] return book (by: May 12 2022 6:00 PM)");
```
love the line break after comma, as well as the 8 spaces before the second line :))

alvinjiang1 commented on src/test/java/duke/helper/ParserTest.java:
```diff
@@ -0,0 +1,56 @@
+        assertEquals(list.getTasks(),
+                listComparison.getTasks());
+
+        //Test mark task
```
Could perhaps replace this comment with javadocs?

alvinjiang1 reviewed (commented):
LGTM! Java coding standards were kept to to a high degree. Could perhaps look into adding javadocs for test classes, but apart from that, great work!!

jehousoh committed:
Fix clear bug

jehousoh committed:
Small tweaks to display

Isaaclhy00 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+
+/**
+ * Main class used to handle inputs
+ */
```
Perhaps can include @Author and @version

Isaaclhy00 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+     * @param dateStr date of the deadline in string
+     * @param dateTime date of the deadline in LocalDate class
+     */
+    public Deadline(String description, String dateStr, LocalDate dateTime) {
```
Is symbol a necessary parameter to be passed in?

Isaaclhy00 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    public static void main(String[] args) {
+        new Duke("./data/duke.txt").run();
```
Perhaps could extract "./" and place them in the Storage/FileWriter class so that users only have to type in "data/duke.txt"

Isaaclhy00 reviewed (commented):
I like that the classes are organized into packages and the Duke class quite closely follows the example given. Overall LGTM :)

jehousoh committed:
Add welcome message

jehousoh committed:
Fix Jar file generation

jehousoh committed:
Add assert to check if index is more than or equals to 0

jehousoh committed:
Clean up code to improve quality

jehousoh committed:
Merge pull request #1 from jehousoh/branch-A-Assertions

Add assertions

jehousoh committed:
Merge branch 'master' of https://github.com/jehousoh/ip

jehousoh committed:
Merge branch 'master' into branch-A-CodeQuality

jehousoh committed:
Merge pull request #2 from jehousoh/branch-A-CodeQuality

Clean up code to improve quality

jehousoh committed:
Merge branch 'master' of https://github.com/jehousoh/ip

jehousoh committed:
Add CI

jehousoh committed:
No way for users to find list of commands

To help the users know what commands are available

Add a help command to allow us to know what commands are available
and how we are able to use them

To provide a seamless integration onto the GUI

jehousoh committed:
Fix 'bye' command to exit after 3 seconds

jehousoh committed:
Fix code quality

jehousoh committed:
Improve the GUI interface

jehousoh committed:
Improved the gui design

jehousoh committed:
Fix JUnit test

jehousoh committed:
Merge branch 'branch-A-BetterGui'

jehousoh committed:
Add invalid time exception

jehousoh committed:
Fix corner case of description of task being invalidTime

jehousoh committed:
Add commandTest and find test

jehousoh committed:
Add user guide

jehousoh committed:
Change phil and user photo

jehousoh committed:
Update README.md

jehousoh committed:
Update readme and Ui photo

jehousoh committed:
Update header comments
